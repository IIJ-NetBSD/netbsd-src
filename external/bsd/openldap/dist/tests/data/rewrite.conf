rewriteEngine on

# We have an attribute value from a DN that we want to paste into a filter
#
# N.B. Do not use ${escape2filter(${unescapedn($1)})}, but chain them inside
# the same rewriteMap like this, since only that is safe in the presence of
# arbitrary data
rewriteMap escape dn2filter unescapedn escape2filter

# We have a DN that we want to paste into a filter
rewriteMap escape dn2dnfilter escape2filter

# We have a filter value and want to construct a DN
rewriteMap escape fitler2dn unescapefilter escape2dn

rewriteContext testdn2filter
rewriteRule "^([^=]*)=([^,+]*)((\+[^,]*)?,(.*))?" "(&($1=${dn2filter($2)})(entryDN:dnOneLevelMatch:=$5))" ":"

rewriteContext testdn2dnfilter
rewriteRule ".*" "entryDN=${dn2dnfilter($0)}" ":"

rewriteContext testfilter2dn
rewriteRule "^\(([^=]*)=([^)]*)\)" "$1=${fitler2dn($2)},dc=example,dc=com" ":"
