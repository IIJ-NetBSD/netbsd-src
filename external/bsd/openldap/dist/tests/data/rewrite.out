# making a filter from a DN
uid=test\20\2c\31\\ -> (&(uid=test ,1\5C)(entryDN:dnOneLevelMatch:=)) [0:ok]
uid=test,ou=People,dc=example,dc=com -> (&(uid=test)(entryDN:dnOneLevelMatch:=ou=People,dc=example,dc=com)) [0:ok]
cn=test\00)(uid=test,dc=example,dc=com -> (&(cn=test\00\29\28uid=test)(entryDN:dnOneLevelMatch:=dc=example,dc=com)) [0:ok]
cn=* -> (&(cn=\2A)(entryDN:dnOneLevelMatch:=)) [0:ok]
cn=*\\ -> (&(cn=\2A\5C)(entryDN:dnOneLevelMatch:=)) [0:ok]
cn=*\ -> (null) [-1:error]

# pasting a DN into a filter
uid=test\20\31\\ -> entryDN=uid=test\5C20\5C31\5C\5C [0:ok]
cn=test)(uid=test,dc=example,dc=com -> entryDN=cn=test\29\28uid=test,dc=example,dc=com [0:ok]
cn=* -> entryDN=cn=\2A [0:ok]
cn=*\\ -> entryDN=cn=\2A\5C\5C [0:ok]
cn=*\ -> entryDN=cn=\2A\5C [0:ok]

# pasting a filter into a DN
(uid=test) -> uid=test,dc=example,dc=com [0:ok]
(cn=something ,\29+\28 \2A=) -> cn=something \2C)\2B( *\3D,dc=example,dc=com [0:ok]
(description=test\20\31*) -> (null) [-1:error]
(description=test\20\31) -> description=test 1,dc=example,dc=com [0:ok]
